### ISSUE TITLE
function dialog says SIGSEGV: Illegal storage access. (Attempt to read from nil?)

### ISSUE THREAD

**Issue by retsyo** [arturo-lang/arturo#681]
for the code `dialog "Select a file to open"`, if I press <kbd>ESC</kbd>, I get `SIGSEGV: Illegal storage access. (Attempt to read from nil?)`. Will it be ok to return `""`?

**Comment by retsyo** [comment]
btw, how to set wildcard, i.e., "*.doc", "*.*".?

**Comment by drkameleon** [comment]
That's a very good catch. I haven't thoroughly tested it, so I'll have to look into it. But, yes, returning an error value (normally `null`) would be a good idea. 😉 

> btw, how to set wildcard, i.e., ".doc", ".*".?

Hmm... Right now, there is no such option. I'll have to look into it as well...

**Comment by drkameleon** [comment]
> for the code dialog "Select a file to open", if I press ESC, I get SIGSEGV: Illegal storage access. (Attempt to read from nil?)

Can you please let me know what OS this is on? (I tried reproducing the error on macOS and I cannot trigger it by any means...)

**Comment by retsyo** [comment]
windows 10 64 bits

**Comment by drkameleon** [comment]
> windows 10 64 bits

Thanks! I'm looking into it!

**Comment by RickBarretto** [comment]
I tested it, if you close using `X` button, you'll get the same.
I think that it's because nothing was returned.

**Comment by RickBarretto** [comment]
I tried to solve it comparing the result with `NULL` and using other approach to convert `std::string` to `cstring`/`char *`, but nothing... I couldn't fix it. 🙁

**Comment by RickBarretto** [comment]
I'm about of 2 hours trying to solve this, but I could not.
So, going to the next one!

**Comment by drkameleon** [comment]
Have you looked if there is an updated pfd implementation (I mean outside of Arturo)? Perhaps, that could be a very welcome surprise! haha

Let me see...

**Comment by RickBarretto** [comment]
> Have you looked if there is an updated pfd implementation (I mean outside of Arturo)?

Yes, first I tried to fix without updating. Looking for where the code was breaking.

I replaced strcpy by a function from strsafe too, but it did not solve anything. 

Then I updated the pfd.cc and updated the header, but nothing changed...

I'd read their documentation, if the user closes the window, it would return an empty string...
